### PR TITLE
Phylotree scale options, bug fixes

### PR DESCRIPTION
**Pull request by florence-77**
### New Feature and Bug Fixes
This commit adds the option to change the horizontal scale in phylotrees to either linear (default) or logarithmic, as we've found that a logarithmic scale sometimes makes phylotrees more user-friendly.

This commit also fixes two bugs in the phylotree code where the specified colors for each organism were not working, and the feature names were not linking correctly. These bugs were fixed by @almasaeed2010.

## Testing
These changes can be tested simply by going to any Phylogenetic Tree (set up via the Chado Newick Tree Loader), and clicking on the blue box in the Tree View tab to view all settings for the tree. The scale can be changed under the setting 'Phylogram Scale', and the colors can be changed under 'Node Colors by Organism'.

## Screenshots (if appropriate):
The first screenshot is an example of a phylotree with a linear scale, and the second is of a phylotree with a logarithmic scale.

Screen Shot 2020-06-30 at 3 29 06 PM

Screen Shot 2020-06-30 at 3 29 38 PM

### CODE REVIEW

**Comment by laceysanderson** [comment]
Thank you for the contribution @noahcaldwell7 and @almasaeed2010! I do not have enough experience with phylotree to review this but I'm tagging @spficklin in the hopes that he will have time.

File: tripal_chado/includes/tripal_chado.phylotree.inc [hunk 1]
```diff
@@ -59,6 +60,7 @@ function tripal_phylogeny_prepare_tree_viewer($phylotree) {
         'interior_node_size' => variable_get('tripal_phylogeny_default_interior_node_size', 1),
         'leaf_node_size' => variable_get('tripal_phylogeny_default_leaf_node_size', 6),
         'skipTicks' => $skip_ticks,
+        'phylogram_scale' => variable_get('tripal_phylogeny_default_phylogram_scale', 6),
```
Comment (florence-77) [inline tripal_chado/includes/tripal_chado.phylotree.inc]: 63: non-typographical addition

File: tripal_chado/includes/tripal_chado.phylotree.inc [hunk 2]
```diff
@@ -319,6 +318,19 @@ function tripal_phylogeny_default_plots_form($form, &$form_state) {
...
+    '#description' => 'Please specify the scale to use.',
+    '#default_value' => variable_get('tripal_phylogeny_default_phylogram_scale', 1),
+    '#options' => array(
+      1 => t('Linear'),
+      2 => t('Logarithmic'),
+    ),
+    '#size' => 2,
+  ];
+
+
```
Comment (florence-77) [inline tripal_chado/includes/tripal_chado.phylotree.inc:333]: 321-333: non-typographical addition

File: tripal_chado/includes/tripal_chado.phylotree.inc [hunk 3]
```diff
@@ -467,6 +481,7 @@ function tripal_phylogeny_default_plots_form_submit($form, &$form_state) {
     variable_set('tripal_phylogeny_default_root_node_size', $form_state['values']['root_node_size']);
     variable_set('tripal_phylogeny_default_interior_node_size', $form_state['values']['interior_node_size']);
     variable_set('tripal_phylogeny_default_leaf_node_size', $form_state['values']['leaf_node_size']);
+    variable_set('tripal_phylogeny_default_phylogram_scale', $form_state['values']['phylogram_scale']);
```
Comment (florence-77) [inline tripal_chado/includes/tripal_chado.phylotree.inc:484]: 484: non-typographical addition

File: tripal_chado/theme/js/d3.phylogram.js [hunk 4]
```diff
@@ -63,219 +63,256 @@
...
       case 4:
-        coordAngle = 3*quarterAngle + coordBaseAngle
+        coordAngle = 3 * quarterAngle + coordBaseAngle
     }
-    return coordAngle
+    return coordAngle;
   }
 
-  function scaleBranchLengths(nodes, w) {
+  function scaleBranchLengths(nodes, w, options) {
```
Comment (florence-77) [inline tripal_chado/theme/js/d3.phylogram.js:174]: 174: non-typographical addition

File: tripal_chado/theme/js/d3.phylogram.js [hunk 5]
```diff
@@ -63,219 +63,256 @@
...
+          .range([0, w]);
+        break;
+      case 2:
+        yscale = d3.scale.log()
+          .domain([0.01, d3.max(rootDists)])
+          .range([0, w]);
+        break;
+      default: // shouldn't happen
+        break;
+    }
```
Comment (florence-77) [inline tripal_chado/theme/js/d3.phylogram.js:204]: 190-204: non-typographical addition

File: tripal_chado/theme/js/d3.phylogram.js [hunk 6]
```diff
@@ -63,219 +63,256 @@
...
+      switch (options.phylogram_scale) {
+        case 1:
+          var yscale = d3.scale.linear()
+            .domain([0, d3.max(rootDists)])
+            .range([0, w]);
+        case 2:
+          var yscale = d3.scale.log()
+            .domain([0.01, d3.max(rootDists)])
+            .range([0, w]);
+      }
```
Comment (florence-77) [inline tripal_chado/theme/js/d3.phylogram.js:256]: 246-256: non-typographical addition

File: tripal_chado/theme/js/tripal_phylogeny.js [hunk 7]
```diff
@@ -40,14 +40,15 @@
   var phylogeny_organism_color = function(d) {
     var organism_color = Drupal.settings.tripal_chado.org_colors;
     var color = null;
-    if (d.genus) {
-      color = organism_color[d.organism_id];
+
+    if (d.fo_genus) {
+      color = organism_color[d.fo_organism_id];
```
Comment (florence-77) [inline tripal_chado/theme/js/tripal_phylogeny.js:45]: 45: non-typographical addition

File: tripal_chado/theme/js/tripal_phylogeny.js [hunk 8]
```diff
@@ -96,7 +97,12 @@
       }
     }
     else {
-      // If this node is not associated with a feature but it has an 
+      if(d.feature_eid) {
+        window.location.href = baseurl + '/bio_data/' + d.feature_eid;
+
+        return;
+      }
```
Comment (florence-77) [inline tripal_chado/theme/js/tripal_phylogeny.js:104]: 100-104: non-typographical addition

File: tripal_chado/theme/js/tripal_phylogeny.js [hunk 9]
```diff
@@ -121,7 +127,8 @@
       'nodeMouseOver' : phylogeny_node_mouse_over,
       'nodeMouseOut' : phylogeny_node_mouse_out,
       'nodeMouseDown' : phylogeny_node_mouse_down,
-      'skipTicks' : tree_options['skipTicks']
+      'skipTicks' : tree_options['skipTicks'],
+      'phylogram_scale' : tree_options['phylogram_scale']
```
Comment (florence-77) [inline tripal_chado/theme/js/tripal_phylogeny.js:131]: 131: non-typographical addition

**Review by florence-77 — COMMENTED** [review]
My IDE, PhpStorm, altered the code formatting a great deal when I used the autoformat hotkey. The formatting is much more up-to-date now, but as a result it's more difficult to review the actual changes to the code. This review adds a comment next to each non-typographical addition to help point reviewers to the important things.

File: tripal_chado/includes/tripal_chado.phylotree.inc (hunk 1 above)
Comment (almasaeed2010) [inline tripal_chado/includes/tripal_chado.phylotree.inc]: Default to 1 (linear) as it was before this PR.
```suggestion
        'phylogram_scale' => variable_get('tripal_phylogeny_default_phylogram_scale', 1),
```

**Comment by bradfordcondon** [comment]
Looks pretty good. 
One issue i had is using the [tripal devseed](https://github.com/statonlab/tripal_dev_seed/blob/master/Fexcel_mini/sequences/mrna_mini.fasta.tree) tree is that the log scale is too close together.  However my feelings on that is if the functionality is good for your dataset we can accept and add a label overlap feature as needed.

Screen Shot 2020-07-22 at 5 49 18 PM

**Review by risharde — APPROVED** [review]
Looks good

**Comment by bradfordcondon** [comment]
thanks for the review @risharde.  Merging with 2 reviews.  Thank you for the submission @noahcaldwell7 good job

**Review by bradfordcondon — APPROVED** [review]

**Pull request merged**